app/models/neighbour_letter.rb

```diff
@@ -3,6 +3,9 @@
 class NeighbourLetter < ApplicationRecord
   belongs_to :neighbour
 
+  validates :resend_reason, absence: true, on: :create, if: -> { neighbour.last_letter_sent_at.nil? }
+  validates :resend_reason, presence: true, on: :create, if: -> { neighbour.last_letter_sent_at.present? }
+
```

Comment:
Could do these as:

``` ruby
validates :resend_reason, absence: true, on: :create, unless: :resend?
validates :resend_reason, presence: true, on: :create, if: :resend?

private

  def resend?
    neighbour.last_letter_sent_at?
  end
```

Should the validations be `on: :create` or do they need to be on `update` as well?

Reply:
I'm not sure if we're ever likely to modify the letter object after creation but it can't hurt to make certain.

Reply:
you can just remove `on`